sequencer/src/api.rs

```diff
@@ -1575,21 +1611,23 @@ mod test {
                 .send()
                 .await
                 .unwrap();
-
+            dbg!(&stop_voting_view);
+            dbg!(&height);
             for peer in &network.peers {
                 let state = peer.consensus().read().await.decided_state().await;
 
                 match state.chain_config.resolve() {
                     Some(cf) => {
-                        if cf != chain_config_upgrade && height as u64 > stop_voting_view {
```

Comment:
isn't the break for the outer loop? . The idea is that when the fee upgrade is complete, we check that the chain config in the validated state is updated. It should always be present in this test, but in reality, if it is not present, it will catch up automatically. If the config is not updated*and the block height is greater than the stop proposing height, we panic.

Reply:
I think the problem is with stop_voting_view. It needs to be updated to take it from the upgrade mode parameter that we are passing. So for the view, it should be height > stop_proposing_view 
and for time it should be current_time > stop_proposing_time. I forgot to update it when we introduced time based parameters.

Reply:
Also, we check chain config upgrade for each node in the network

Reply:
Yes `break` is for outer loop. As soon as we iterate through the first set of peers the loop exits. I think we need to `break` on a successful match of the two chain_configs like in 97b377d38f074ce49c239d840f1c025b897316e6. I'm not sure just yet how to test all the peers.

Reply:
I think 5050a85787a01c3ea02da5733f8ddad7e86158bf is what we discused.